src/VisualStudio/Xaml/Impl/Implementation/XamlProject.cs

```diff
@@ -12,20 +14,21 @@ namespace Microsoft.VisualStudio.LanguageServices.Xaml
     [DebuggerDisplay("{GetDebuggerDisplay(),nq}")]
     internal class XamlProject : AbstractLegacyProject
     {
-        public XamlProject(IVsHierarchy hierarchy, IServiceProvider serviceProvider, VisualStudioWorkspaceImpl visualStudioWorkspace) :
+        public XamlProject(VisualStudioProjectTracker projectTracker, IVsHierarchy hierarchy, IServiceProvider serviceProvider, VisualStudioWorkspaceImpl visualStudioWorkspace) :
             base(
-                visualStudioWorkspace.ProjectTracker,
+                projectTracker,
                 reportExternalErrorCreatorOpt: null,
-                projectSystemName: $"{XamlProject.GetProjectName(hierarchy)}|{nameof(XamlProject)}",
+                projectSystemName: $"{XamlProject.GetProjectName(hierarchy)}-{nameof(XamlProject)}",
                 hierarchy: hierarchy,
                 language: StringConstants.XamlLanguageName,
                 serviceProvider: serviceProvider,
                 visualStudioWorkspaceOpt: visualStudioWorkspace,
                 hostDiagnosticUpdateSourceOpt: null)
         {
-            // We initialized ProjectSystemName to include nameof(XamlProject).
-            // Update DisplayName to just the project name.
-            UpdateProjectDisplayName(XamlProject.GetProjectName(hierarchy));
+            // Use default options
+            SetOptions(new CSharpCompilationOptions(OutputKind.ConsoleApplication), new CSharpParseOptions());
```

Comment:
I think I would prefer we instead fix whatever felt this was necessary. My worry is this might cause other things to activate as well.